dsc_lib/src/discovery/command_discovery.rs

```diff
@@ -37,31 +63,74 @@ impl CommandDiscovery {
         }
     }
 
+    fn get_resource_path_setting() -> Result<ResourcePathSetting, DscError>
+    {
+        if let Ok(v) = get_setting("resourcePath") {
+            // if there is a policy value defined - use it; otherwise use setting value
+            if v.policy != serde_json::Value::Null {
+                match serde_json::from_value::<ResourcePathSetting>(v.policy) {
+                    Ok(v) => {
+                        return Ok(v);
+                    },
+                    Err(e) => { return Err(DscError::Setting(format!("{e}"))); }
+                }
+            } else if v.setting != serde_json::Value::Null {
+                match serde_json::from_value::<ResourcePathSetting>(v.setting) {
+                    Ok(v) => {
+                        return Ok(v);
+                    },
+                    Err(e) => { return Err(DscError::Setting(format!("{e}"))); }
+                }
+            }
+        }
+
+        Err(DscError::Setting("Could not read 'resourcePath' setting".to_string()))
+    }
+
     fn get_resource_paths() -> Result<Vec<PathBuf>, DscError>
     {
+        let mut resource_path_setting = ResourcePathSetting::default();
+
+        match Self::get_resource_path_setting() {
+            Ok(v) => {
+                resource_path_setting = v;
+            },
+            Err(e) => { 
+                debug!("{e}");
+            }
+        }
+
         let mut using_custom_path = false;
+        let mut paths: Vec<PathBuf> = vec![];
 
-        // try DSC_RESOURCE_PATH env var first otherwise use PATH
-        let path_env = if let Some(value) = env::var_os("DSC_RESOURCE_PATH") {
+        let dsc_resource_path = env::var_os("DSC_RESOURCE_PATH");
+        if resource_path_setting.allow_env_override && dsc_resource_path.is_some(){
+            let value = dsc_resource_path.unwrap();
             debug!("Using DSC_RESOURCE_PATH: {:?}", value.to_string_lossy());
             using_custom_path = true;
-            value
+            paths.append(&mut env::split_paths(&value).collect::<Vec<_>>());
         } else {
-            trace!("DSC_RESOURCE_PATH not set, trying PATH");
-            match env::var_os("PATH") {
-                Some(value) => {
-                    trace!("Original PATH: {:?}", value.to_string_lossy());
-                    value
-                },
-                None => {
-                    return Err(DscError::Operation("Failed to get PATH environment variable".to_string()));
+            for p in resource_path_setting.directories {
+                let v = PathBuf::from_str(&p);
+                paths.push(v.unwrap_or_default());
```

Comment:
What is default here?  It may be better to error if it can't unwrap.

Reply:
There was an error path originally;
but this line was giving me a headache in the style check, because, as it turns out, `from_str` can never fail despite its signature.